Dockerfile

```diff
@@ -41,9 +41,9 @@ RUN mkdir -p /app \
 WORKDIR /app
 
 RUN apt-get update && apt-get install -y curl make gcc g++
-ENV GOLANG_VERSION 1.15.5
-ENV GOLANG_DOWNLOAD_SHA256 9a58494e8da722c3aef248c9227b0e9c528c7318309827780f16220998180a0d
-ENV GOLANG_DOWNLOAD_URL https://golang.org/dl/go$GOLANG_VERSION.linux-amd64.tar.gz
+ENV GOLANG_VERSION "1.16.3"
```

Comment:
Should we go to `1.16.4` (released yesterday)?

Reply:
Hmmm.... It's not super realistic for us to keep up-to-date with minor version releases. I'd prefer to drop the minor version entirely at this point (ie; `ENV GOLANG_VERSION "1.16"`). WDYT?

Reply:
Yeah, I was thinking the same, re: churn with minor version updates. I like that idea.

Reply:
After further investigation I don't think pulling the major version is the best solution. 

1. There's no checksum available for the major version mirror (please share if you find)
2. Given the [go promise of compatability](https://golang.org/doc/go1compat) I don't see much of a reason in adjusting the minor version 
3. We don't use any of the features in the latest minor version release

I'd prefer if we find a proper solution to grabbing the latest release, which is out of scope for this PR. Likely we'll be pulling & building from source, we can discuss in a dedicated issue if needed.

Reply:
>I'd prefer if we find a proper solution to grabbing the latest release, which is out of scope for this PR. Likely we'll be pulling & building from source, we can discuss in a dedicated issue if needed.

This makes sense.

I do see that 1.16.4 includes a security fix for the `net/http` package, though I haven't dug into the specifics.

https://golang.org/doc/devel/release.html#go1.16

> go1.16.4 (released 2021/05/06) includes a security fix to the net/http package, as well as bug fixes to the runtime, the compiler, and the archive/zip, time, and syscall packages. See the Go 1.16.4 milestone on our issue tracker for details.